### ISSUE TITLE
Network stats are not sampled correct

### ISSUE THREAD

**Issue by ChristianKniep** [Yelp/fullerite#198]
Hey there,

the sampled network stats are not valid anymore - I suppose.
When sampling the stats for a given container, the network stats are now exposed per interface.
```
$ curl -s --unix-socket /var/run/docker.sock http:/containers/carbon/stats|jq .
*snip*
  "networks": {
    "eth0": {
      "rx_bytes": 442423987351,
      "rx_packets": 9150523,
      "rx_errors": 0,
      "rx_dropped": 0,
      "tx_bytes": 295780357,
      "tx_packets": 4407470,
      "tx_errors": 0,
      "tx_dropped": 0
    }
  }
}
```
The current code assumes to find a key `network`, which sums up the usage over all interfaces.
With API `v1.22` this seems not to be valid anymore.

But that might be something for the dockerclient library...

**Comment by tsheasha** [comment]
Hey @ChristianKniep , indeed this seems to not be the case, and a fix was introduced to the docker client library with https://github.com/fsouza/go-dockerclient/commit/adc42c2863dd78dcabe6b50b649a7ac1cee8b5c2
which was then followed up by another patch for backwards compatibility: https://github.com/fsouza/go-dockerclient/commit/02a8beb401b20e112cff3ea740545960b667eab1

However we have pinned a certain commit in our Gomfile for this library: https://github.com/Yelp/fullerite/blob/master/Gomfile#L7

Which is a commit dated the 10th of December 2015, that precedes either of the aforementioned commits. If you want to go ahead and update the Gomfile with and pin commit https://github.com/fsouza/go-dockerclient/commit/02a8beb401b20e112cff3ea740545960b667eab1 or after that would be awesome!

**Comment by ChristianKniep** [comment]
Dealt with in #199

**Comment by ChristianKniep** [comment]
Please keep it open, since I see gaps in between the samplings and I want to understand this.
I would have expected, that fullerite samples the same amount of metrics each time it runs - but that's somehow not the case. :/
```
bash-4.3# tail -f /var/log/supervisor/fullerite.log
time="09 Mar 16 15:53 UCT" level=info msg="Starting to emit 9 metrics" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:53 UCT" level=info msg="POST of 9 metrics to Graphite took 0.002851 seconds" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:54 UCT" level=info msg="Starting to emit 9 metrics" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:54 UCT" level=info msg="POST of 9 metrics to Graphite took 0.015013 seconds" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:54 UCT" level=info msg="Starting to emit 7 metrics" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:54 UCT" level=info msg="POST of 7 metrics to Graphite took 0.002433 seconds" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:54 UCT" level=info msg="Starting to emit 45 metrics" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:54 UCT" level=info msg="POST of 45 metrics to Graphite took 0.010613 seconds" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:55 UCT" level=info msg="Starting to emit 32 metrics" app=fullerite handler=Graphite pkg=handler
time="09 Mar 16 15:55 UCT" level=info msg="POST of 32 metrics to Graphite took 0.004707 seconds" app=fullerite handler=Graphite pkg=handler
```
The config...
```
$ cat /etc/fullerite/fullerite.conf
{
    "prefix": "fullerite.",
    "interval": 10,
    "fulleritePort": 19191,
    "internalServer": {"port":"29090","path":"/metrics"},
    "collectorsConfigPath": "/etc/fullerite/conf.d",

    "collectors": ["DockerStats"],

    "handlers": {
        "Graphite": {
            "server": "carbon.service.consul",
            "port": "2003",
            "interval": "10",
            "max_buffer_size": 300,
            "timeout": 2
        }
    }
}
```

**Comment by tsheasha** [comment]
hey @ChristianKniep , fullerite *does* get the same number of collected metrics each time it runs, and what you see in the logs is not saying anything different. The reason is, each collector has a dedicated channel to the handler, this means as soon as a collection happens, the handler gets the metrics and assigns them to a bucket for that collector. Each time the number of metrics in the bucket exceeds $max_buffer_size of interval of emission,. this is flushed to the backend, in your case Graphite.

So basically what you see in your logs is:
time="09 Mar 16 15:53 UCT" level=info msg="POST of 9 metrics to Graphite took 0.002851 seconds" app=fullerite handler=Graphite pkg=handler     ====> this represent the emission of the metrics from one collector

time="09 Mar 16 15:54 UCT" level=info msg="Starting to emit 45 metrics" app=fullerite handler=Graphite pkg=handler ===> this represents another collector and so on and so forth

**Comment by ChristianKniep** [comment]
Hey @tsheasha,

ok, that make sense, but since time-series data is timestamped there should not be any loss involved, right? I'll have another look tomorrow; still fighting my jet-lack. :)
